mlir/lib/Dialect/Vector/Transforms/LowerVectorTransfer.cpp

```diff
@@ -201,12 +205,19 @@ struct TransferWritePermutationLowering
     // Generate new transfer_write operation.
     Value newVec = rewriter.create<vector::TransposeOp>(
         op.getLoc(), op.getVector(), indices);
+
+    auto vectorType = cast<VectorType>(newVec.getType());
+
+    if (vectorType.isScalable() && !*vectorType.getScalableDims().end()) {
```

Comment:
I believe `*vectorType.getScalableDims().end()` is could be an out-of-bounds access. The `.end()` method returns an iterator passed the end of the array, I think what you probably want is: 

```suggestion
    if (vectorType.isScalable() && !vectorType.getScalableDims().back()) {
```

Reply:
Btw, for this check there's `isLegalVectorType()` here: https://github.com/llvm/llvm-project/blob/main/mlir/lib/Dialect/ArmSME/Transforms/VectorLegalization.cpp#L421-L430

Maybe this could be moved to some general utils?

Reply:
> Btw, for this check there's isLegalVectorType() 

Re-using that hook would make sense to me, but we can't call it `isLegalVectorType` (and, in general, we need to be careful when labeling vectors as "illegal"):
* vectors like `vector<[4]x4xi32>` are perfectly legal at the "abstract" Vector dialect level,
* only once we start lowering to LLVM (and/or SVE/SME), these types needs to be eliminated and it makes sense to consider them as "illegal".

For scalable vectors that have only 1 scalable dim, this code is correct . @cfRod, I suggest adding a comment that we are assuming that at most 1 dim is scalable. @MacDue , I can rename and move `isLegalVectorType` in a separate PR. WDYT?

Reply:
Fine with me :+1:  (I agree with the naming, the `isLegal` was only within the context of the ArmSME pass).